db/version_set.h

```diff
@@ -130,6 +130,38 @@ class VersionStorageInfo {
 
   void AddFile(int level, FileMetaData* f);
 
+  // Resize/Initialize the space for compact_cursor_
+  void ResizeCompactCursors(int level) {
+    compact_cursor_.resize(level, InternalKey());
+  }
+
+  const std::vector<InternalKey>& GetCompactCursors() const {
+    return compact_cursor_;
+  }
+
+  // REQUIRES: ResizeCompactCursors has been called
+  void AddCursorForOneLevel(int level,
+                            const InternalKey& smallest_uncompacted_key) {
+    compact_cursor_[level] = smallest_uncompacted_key;
+  }
+
+  // REQUIRES: lock is held
+  // Update the compact cursor and advance the file index so that it can point
+  // to the next cursor
+  const InternalKey& GetNextCompactCursor(int level) {
+    int cmp_idx = ++next_file_to_compact_by_size_[level];
```

Comment:
Do we need to mutate `next_file_to_compact_by_size_`?

Reply:
In compaction_picker_level.cc, it starts from `next_file_to_compact_by_size_[level]` to pick files. If we do not change it, it is possible that under round-robin policy, we always return false when the first file is chosen to be compacted in the last compaction.

Reply:
Yeah, I agree with you to make the initial compaction simpler. I will just change it into `next_file_to_compact_by_size_[level]+1` for now.